src/main/java/org/folio/circulation/services/LostItemFeeChargingService.java

```diff
@@ -142,9 +142,12 @@ private UUID getOwnerServicePoint(Loan loan) {
   }
 
   private CompletableFuture<Result<ReferenceDataContext>> fetchFeeFineOwner(ReferenceDataContext referenceData) {
-    final String servicePointId = getOwnerServicePoint(referenceData.loan).toString();
+    final String permanentLocationId = referenceData.loan.getItem().getPermanentLocationId();
 
-    return feeFineOwnerRepository.findOwnerForServicePoint(servicePointId)
+    return locationRepository.fetchLocationById(permanentLocationId)
+      .thenApply(r -> r.map(Location::getPrimaryServicePointId))
+      .thenApply(r -> r.map(UUID::toString))
+      .thenCompose(r -> r.after(feeFineOwnerRepository::findOwnerForServicePoint))
```

Comment:
Could this mapping from UUID to String either be avoided or moved to an overload in the fee fine owner repository?

Reply:
Added an override method.